src/Microsoft.AspNet.Authentication.OpenIdConnect/OpenidConnectAuthenticationHandler.cs

```diff
@@ -235,180 +276,215 @@ protected override AuthenticationTicket AuthenticateCore()
             {
                 IFormCollection form = await Request.ReadFormAsync();
```

Comment:
My remark was rather about the `Request.Body.Seek(0, SeekOrigin.Begin);` call, which is not necessary if `Request.ReadFormAsync` caches the result and doesn't need to read the request stream again (which wouldn't work for streamed requests).